lib/ember_cli/runner.rb

```diff
@@ -32,9 +35,11 @@ def run!(command)
 
     private
 
-    def write(output, streams:)
-      streams.each do |stream|
-        stream.write(output)
+    def redirect(stream, pipes:)
+      Tee.open(*pipes, stdout: nil) do |pipe|
+        while line = stream.gets
```

Comment:
I don't know the `tee` gem or too much about the nuances Ruby's process-running features, so it's hard to weigh in on the general approach.

It's not **generally** a problem with blocking, as `STDOUT` and `STDERR` have pretty large buffers which protect writers from being blocked. But it is possible.

This will also stream all `STDOUT` messages first, then all `STDERR` messages, which may be unexpected.